File: DCP-03-25/Merge-Two-2D-Arrays-by-Summing-Values.cs
public class Solution {
    public int[][] MergeArrays(int[][] nums1, int[][] nums2) => nums1
        .Concat(nums2)
        .GroupBy(num => num[0])
        .Select(g => g.Aggregate(new int[2], (sum, item) => new [] { item[0], sum[1] + item[1] }))
        .OrderBy(item => item[0])
        .ToArray();
}

File: DCP-03-25/Merge-Two-2D-Arrays-by-Summing-Values.py
class Solution:
    def mergeArrays(self, nums1: List[List[int]], nums2: List[List[int]]) -> List[List[int]]:
        s1 = dict(nums1)
        s2 = dict(nums2)
        merge = Counter(s1) + Counter(s2)
        merge = sorted(merge.items())
        res = list(merge)
        return res
